src/main/webapp/app/course/competencies/competency-management/competency-management-table.component.html

```diff
@@ -0,0 +1,116 @@
+                <button class="btn" id="competencyImportAllButton" (click)="openImportAllModal()" ngbDropdownItem>
+                    <fa-icon [icon]="faFileImport" />
+                    <span>{{ 'artemisApp.' + competencyType + '.manage.importAllButton' | artemisTranslate }}</span>
+                </button>
+                <a class="btn" [routerLink]="['/course-management', courseId, competencyType + '-management', 'import']" ngbDropdownItem>
+                    <fa-icon [icon]="faFileImport" />
+                    <span>{{ 'artemisApp.' + competencyType + '.manage.importFromCoursesButton' | artemisTranslate }}</span>
+                </a>
+                @if (standardizedCompetenciesEnabled) {
+                    <a class="btn" [routerLink]="['/course-management', courseId, competencyType + '-management', 'import-standardized']" ngbDropdownItem>
+                        <fa-icon [icon]="faFileImport" />
+                        <span>{{ 'artemisApp.' + competencyType + '.manage.importStandardizedButton' | artemisTranslate }}</span>
+                    </a>
+                }
+            </div>
+        </div>
+        <a class="btn btn-primary" [routerLink]="['/course-management', courseId, competencyType + '-management', 'create']">
+            <fa-icon [icon]="faPlus" />
+            <span>{{ 'artemisApp.' + competencyType + '.manage.createButton' | artemisTranslate }}</span>
+        </a>
+    </div>
+</div>
+<div class="container-fluid overflow-auto">
+    @if (courseCompetencies.length) {
+        <table class="table table-striped">
+            <thead>
+                <tr>
+                    <th class="col">
+                        <span jhiTranslate="global.field.id"></span>
+                    </th>
+                    <th class="col">
+                        <span jhiTranslate="artemisApp.competency.title"></span>
+                    </th>
+                    <th class="col">
+                        <span jhiTranslate="artemisApp.competency.description"></span>
+                    </th>
+                    <th class="col">
+                        <span jhiTranslate="artemisApp.competency.taxonomy"></span>
+                    </th>
+                    <th class="d-none col-lg d-lg-table-cell">
+                        <span jhiTranslate="artemisApp.competency.softDueDate"></span>
+                    </th>
+                    <th class="d-none col-lg d-lg-table-cell">
+                        <span jhiTranslate="artemisApp.competency.masteredStudents"></span>
+                    </th>
+                    <th class="col">
+                        <span jhiTranslate="artemisApp.competency.optional"></span>
+                    </th>
+                    <th class="col"></th>
+                </tr>
+            </thead>
+            <tbody>
+                @for (competency of courseCompetencies; track competency.id; let i = $index) {
```

Comment:
you could just us the for loop without checking the array length and use the @empty block of the for block

similar to
```suggestion
                // HTML of the if
                @for (competency of courseCompetencies; track competency.id; let i = $index) {
                    // HTML of the for
                } @empty {
                    // HTML if list is empty (HTML which is now in the @else block)
                }
```

Reply:
We currently do not even show the table if there is no competency. So your suggestion would at least change the user interface.
Regardless I would consider this out of scope, since this was already implemented this way for competencies before this PR

Reply:
If you would use @empty the table would still not be displayed you could just get rid of the @if(courseCompetencies.length) in line 33.
But if that's out of scoped fine for me 👍